cmds/core/sluinit/uinit.go

```diff
@@ -73,13 +75,38 @@ func main() {
 	}
 	slaunch.Debug("Collectors completed")
 
-	slaunch.Debug("********Step 4: Write eventlog to /boot partition*********")
-	if e := p.EventLog.Persist(); e != nil {
-		log.Printf("EventLog.Persist() failed err=%v", e)
-		os.Exit(1)
+	slaunch.Debug("********Step 4: Measuring target kernel, initrd ********")
+	if e := p.Launcher.MeasureKernel(tpmDev); e != nil {
+		log.Printf("Launcher.MeasureKernel failed err=%v", e)
+		return
 	}
 
-	slaunch.Debug("********Step 5: Launcher called ********")
-	err = p.Launcher.Boot(tpmDev)
-	log.Printf("Boot failed. err=%s", err)
+	slaunch.Debug("********Step 5: Parse eventlogs *********")
+	if e := p.EventLog.Parse(); e != nil {
+		log.Printf("EventLog.Parse() failed err=%v", e)
+		return
+	}
+
+	slaunch.Debug("*****Step 6: Dump logs to disk *******")
+	if e := slaunch.ClearPersistQueue(); e != nil {
+		log.Printf("ClearPersistQueue failed err=%v", e)
+		return
+	}
+
+	slaunch.Debug("********Step *: Unmount all ********")
+	slaunch.UnmountAll()
+
+	slaunch.Debug("********Step 7: Launcher called to Boot ********")
+	if err := p.Launcher.Boot(tpmDev); err != nil {
+		log.Printf("Boot failed. err=%s", err)
+		return
+	}
+}
+
+// unmountAndExit is called on error and unmounts all devices.
+// sluinit ends here.
+func unmountAndExit() {
+	slaunch.UnmountAll()
+	time.Sleep(5 * time.Second) // let queued up debug statements get printed
```

Comment:
are you seeing this problem on real hardware?

Reply:
Yes, This is what i brought up on the slack channel as well. Not only did i see it on my X7 machine, QA team also saw it on X7. Spec sheet here: https://www.oracle.com/technetwork/database/exadata/exadata-x7-2-ds-3908482.pdf